tests/e2e/democracy_test.go

```diff
@@ -253,6 +263,119 @@ func (s *ConsumerDemocracyTestSuite) TestDemocracyRewarsDistribution() {
+	oldAuthParamValue := authKeeper.GetParams(s.consumerCtx()).MaxMemoCharacters
+	oldMintParamValue := mintKeeper.GetParams(s.consumerCtx()).InflationMax
+	s.consumerChain.NextBlock()
+	//at this moment, proposal is executed or deleted if forbidden
+	currentAuthParamValue := authKeeper.GetParams(s.consumerCtx()).MaxMemoCharacters
+	currentMintParamValue := mintKeeper.GetParams(s.consumerCtx()).InflationMax
+	//check that parameters are not changed, since the proposal contained both forbidden and allowed changes
+	s.Assert().Equal(oldAuthParamValue, currentAuthParamValue)
+	s.Assert().NotEqual(newAuthParamValue, currentAuthParamValue)
+	s.Assert().Equal(oldMintParamValue, currentMintParamValue)
+	s.Assert().NotEqual(newMintParamValue, currentMintParamValue)
+	//deposit is refunded
+	s.Assert().Equal(votersOldBalances, getAccountsBalances(s.consumerCtx(), bankKeeper, bondDenom, votingAccounts))
+
+	//submit proposal with allowed changes
+	paramChange = proposaltypes.ParameterChangeProposal{Changes: []proposaltypes.ParamChange{allowedChange}}
+	err = submitProposalWithDepositAndVote(govKeeper, s.consumerCtx(), paramChange, votingAccounts, depositAmount)
+	s.Assert().NoError(err)
+	s.consumerChain.CurrentHeader.Time = s.consumerChain.CurrentHeader.Time.Add(votingParams.VotingPeriod)
+	s.consumerChain.NextBlock()
+	oldMintParamValue = mintKeeper.GetParams(s.consumerCtx()).InflationMax
+	s.consumerChain.NextBlock()
+	currentMintParamValue = mintKeeper.GetParams(s.consumerCtx()).InflationMax
+	//check that parameters are changed, since the proposal contained only allowed changes
+	s.Assert().Equal(newMintParamValue, currentMintParamValue)
+	s.Assert().NotEqual(oldMintParamValue, currentMintParamValue)
+	//deposit is refunded
+	s.Assert().Equal(votersOldBalances, getAccountsBalances(s.consumerCtx(), bankKeeper, bondDenom, votingAccounts))
+
+	//submit proposal with forbidden changes
+	paramChange = proposaltypes.ParameterChangeProposal{Changes: []proposaltypes.ParamChange{forbiddenChange}}
+	err = submitProposalWithDepositAndVote(govKeeper, s.consumerCtx(), paramChange, votingAccounts, depositAmount)
+	s.Assert().NoError(err)
+	s.consumerChain.CurrentHeader.Time = s.consumerChain.CurrentHeader.Time.Add(votingParams.VotingPeriod)
+	s.consumerChain.NextBlock()
+	oldAuthParamValue = authKeeper.GetParams(s.consumerCtx()).MaxMemoCharacters
+	s.consumerChain.NextBlock()
+	currentAuthParamValue = authKeeper.GetParams(s.consumerCtx()).MaxMemoCharacters
+	//check that parameters are not changed, since the proposal contained forbidden changes
+	s.Assert().Equal(oldAuthParamValue, currentAuthParamValue)
+	s.Assert().NotEqual(newAuthParamValue, currentAuthParamValue)
+	//deposit is refunded
+	s.Assert().Equal(votersOldBalances, getAccountsBalances(s.consumerCtx(), bankKeeper, bondDenom, votingAccounts))
+}
+
+func (s *ConsumerDemocracyTestSuite) TestDemocracyGovernanceWhitelistingKeys() {
```

Comment:
This looks like a unit test, but is there an advantage to using `ConsumerDemocracyTestSuite`?

Reply:
Thanks for pointing that out. There's no need for ConsumerDemocracyTestSuite, since we need only consumer app with the initialized param subspaces to check if the keys and subspace string exists. The test is moved from e2e folder.